segments-guide.md

```diff
@@ -0,0 +1,327 @@
+
+## Best Practices
+
+Please follow these guidelines when using the Segments APIs:
+
+*  Make multiple, smaller requests instead of a large, single request.
+*  Request data once and cache it.
+*  Use caution when updating a segment so that you do not alter the original segment used by others.
+*  Avoid creating duplicate segments with the same definition. Creating many segments will affect performance for your company in some situations.
+*  Use the Analytics UI for creating segments from scratch, if possible. Because of the flexibility in using segments, the UI offers significant advantages for managing and optimizing the complexity of this task. Consider creating a segment in the UI that matches, as closely as possible, the segment you want, then modifying the JSON so that it does what you want programmatically. Then you can `POST` the JSON body of that modified segment to the API.
+
+## /segments Endpoint Description
+
+The `/segments` endpoint description is shown in our [Swagger UI](https://adobedocs.github.io/analytics-2.0-apis/). Use the Swagger UI to see endpoint summaries, available methods, parameters, example values, models, and status codes, and to try out the API.
+
+## Templates
+
+Adobe provides many predefined segments, or templates, for clients in specific situations. Although original templates cannot be modified, copies of them can be made and then the copies can be modified. If you want to work with segments templates programmatically, you can identify them by the additional attribute `template: true` as shown at the bottom of the following example:
+
+```
+
+{
+      "id": "Has_An_Action",
+      "name": "Has an Action",
+      "description": null,
+      "definition": {
+        "container": {
+          "func": "container",
+          "pred": {
+            "val": {
+              "func": "attr",
+              "name": "variables/mobileaction"
+            },
+            "func": "exists",
+            "description": "Action Name"
+          },
+          "context": "hits"
+        },
+        "func": "segment",
+        "version": [
+          1,
+          0,
+          0
+        ]
+      },
+      "template": true
+},
+
+```
+For more information on segments templates, see [Adobe Help](https://marketing.adobe.com/resources/help/en_US/analytics/segment/seg_build.html).
+
+### Expansions
+
+Segments endpoints support the URL query parameter `expansion`. This parameter allows specifying additional data fields to be populated in response objects. Available expansions are shown in the [Swagger UI](https://adobedocs.github.io/analytics-2.0-apis/) description. The following table describes expansion fields:
+
+|           Field       | Description                                                                                                          |
```

Comment:
The markdown on tables is easier to read and manage in code if you don't put in as many dashes and white space characters or if you go to the trouble to line them up in a mono spaced font.
https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#tables

Reply:
There was just a bunch of white space and what looked like unnecessary spaces in the table. It's minor since it doesn't affect the rendering, but it is easier to scan and read from a code perspective if you pay attention to the columns lining up and don't make the lines a lot longer than needed.

Reply:
Okay. Thanks.